src/backend/PortabilityService.AnalysisEngine/Dockerfile

```diff
@@ -0,0 +1,16 @@
+### Build Stage
+FROM microsoft/aspnetcore-build:2.0.7-2.1.105 AS build
+WORKDIR /src
+
+COPY NuGet.config .
+COPY ./src .
+WORKDIR /src/backend/PortabilityService.AnalysisEngine
+RUN dotnet publish PortabilityService.AnalysisEngine.csproj -c Release -o /app
+
```

Comment:
@mjrousos It looks like we just need 'publish' command which will automatically restore and build before publishing. Is there benefit to have separate steps (better for debugging issue maybe)?

Reply:
Separating restore and build is beneficial because that way the output of each step will be a separate image layer. Then, in the future, if a change is made that requires a re-build (but not a re-restore), `docker build` can re-use the layer containing restore output and not have to spend time restoring NuGet packages again.

In our case, I think separating build and publish probably doesn't really matter (I just copied the Dockerfile format from a template).